addons/website/models/theme_models.py

```diff
@@ -103,20 +103,29 @@ class ThemeMenu(models.Model):
     page_id = fields.Many2one('theme.website.page', ondelete='cascade')
     new_window = fields.Boolean('New Window')
     sequence = fields.Integer()
-    parent_id = fields.Many2one('theme.website.menu', index=True, ondelete="cascade")
+    parent_id = fields.Reference(index=True, ondelete="cascade", selection=[
+        ('website.menu', 'website.menu'), ('theme.website.menu', 'theme.website.menu')
+    ])
```

Comment:
Not a huge fan of the `Reference` field type (maybe I am wrong but that does not seem like a good practice to me, I am afraid of the magic which is behind :grimacing:). Actually with your suggested implementation I think that if someone defines a parent_id which points to a website.menu which is not the default one, you ignore it. The usecase may not be something we want, but the use of Reference makes me think this is authorized.

Maybe to ease the code and its external use, the parent_id field should be just like before... but we add an *extra* boolean field `use_main_menu_as_parent`,  default to True. That way, the main use case is easy: you leave that field to true, you set no `parent_id` and you have a menu in your website. In case you want something more complex (adding a submenu, adding an extra top menu for some complex custo, ...), then you have to set the boolean to false and/or use the parent_id field (or not). And that way we ensure it's more robust as we don't allow a Reference to something which is generic (since we can only refer to theme.website.menu).

Not sure, what do you think?

And for stable I don't know... if I am not mistaken, it was simply impossible to create a menu attached to the main website menu as if you set no parent_id, it was created as an extra top menu (generic :/) right? So maybe we could simulate that boolean (when there is no parent_id) with something like:
- Use website main menu if sequence > 1000
- Just create a new top menu attached to nothing otherwise :shrug: 

That would most likely leave working custo use "extra top menu" (unless they went with huge sequence for some strange reason), and allow 14.0 themes to attach menus to website by setting huge sequences... :shrug: Or we create an extra module like you said, with the boolean I suggested, but that does not really seem better? Although I guess community themes would just have to add it as a dependency? :thinking:

Reply:
> Although I guess community themes would just have to add it as a dependency? thinking

Yes exactly so that's kinda stable at the end as it would work out of the box if the theme module is doing that.

> Actually with your suggested implementation I think that if someone defines a parent_id which points to a website.menu which is not the default one, you ignore it.

That's indeed a missing part of this PR, AFAIC this should also work? I actually wanted to move one of the menu below the `Home` menu to see if the update was behaving correctly. Too bad I didn't or I would have noticed that.
I am pretty sure we want to support that too, is a theme want to create 2 menus below `Contact us` like `Get in touch` and `How to travel to our offices`, or another exemple: `Contact our EU offices` and `Contact our US offices`

I guess the rest of the discussion will depend on what we choose here :o 

By the way, the `reference` field is already what we use for the `theme.ir.ui.view.inherit_id`, so it is not really that frightening, especially since `theme.ir.ui.view` are our main theme record.